docs/core/diagnostics/microsoft-diagnostics-netcore-client.md

```diff
@@ -0,0 +1,459 @@
+using System.Diagnostics.Tracing;
+
+public void TriggerDumpOnCpuUsage(int processId, int threshold)
+{
+    var providers = new List<EventPipeProvider>()
+    {
+        new EventPipeProvider(
+            "System.Runtime",
+            EventLevel.Informational,
+            (long)ClrTraceEventParser.Keywords.None,
+            new Dictionary<string, string>() {
+                { "EventCounterIntervalSec", "1" }
+            }
+        )
+    };
+    var client = new DiagnosticsClient(processId);
+    using(var session = client.StartEventPipeSession(providers))
+    {
+        var source = new EventPipeEventSource(session.EventStream);
+        source.Dynamic.All += (TraceEvent obj) =>
+        {
+            if (obj.EventName.Equals("EventCounters"))
+            {
+                IDictionary<string, object> payloadVal = (IDictionary<string, object>)(obj.PayloadValue(0));
+                IDictionary<string, object> payloadFields = (IDictionary<string, object>)(payloadVal["Payload"]);
+                if (payloadFields["Name"].ToString().Equals("cpu-usage"))
+                {
+                    double cpuUsage = Double.Parse(payloadFields["Mean"].ToString());
+                    if (cpuUsage > (double)threshold)
+                    {
+                        client.WriteDump(DumpType.Normal, "/tmp/minidump.dmp");
+                    }
+                }
+            }
+        };
+        try
+        {
+            source.Process();
+        }
+        catch (Exception) {}
+    }
+}
+```
+
+### Trigger a CPU trace for given number of seconds
+
+This sample shows an example where we trigger an EventPipe session for certain period of time, with the default CLR trace keyword as well as the sample profiler, and read from the stream that gets created as a result and write the bytes out to a file. Essentially this is what `dotnet-trace` uses internally to write a trace file.
+
+```cs
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.NETCore.Client;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Task;
+
+public void TraceProcessForDuration(int processId, int duration, string traceName)
+{
+    var cpuProviders = new List<EventPipeProvider>()
+    {
+        new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Informational, (long)ClrTraceEventParser.Keywords.Default),
+        new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", EventLevel.Informational, (long)ClrTraceEventParser.Keywords.None)
+    };
+    var client = new DiagnosticsClient(processId);
+    using (var traceSession = client.StartEventPipeSession(cpuProviders))
+    {
+        Task copyTask = Task.Run(async () =>
+        {
+            using (FileStream fs = new FileStream(traceName, FileMode.Create, FileAccess.Write))
+            {
+                await traceSession.EventStream.CopyToAsync(fs);
+            }
+        });
+
+        copyTask.Wait(duration * 1000);
```

Comment:
Why is this `copyTask.Wait`? As long as the session is not gone, this will always block for the full duration. You sent the call to a background thread, and then effectively blocked this thread. Is this the best example that we can give here? A Task.WhenAny + Task.Delay may be better, and we should add something to observe the exceptions in `copyTask`.